### PR TITLE
fix ordersDB methods to take correct args

### PR DESCRIPTION
**Pull request by JessicaGreben**
What/Why: 
All methods in orders.go are fixed so that they take projectID and bucketID as separate arguments with appropriate types instead of parsing bucketID in orderDB methods. Also fixed accountingDB where it had this same anti-pattern. 

Please describe the tests:
Updated tests to take projectID and bucketID as separate arguments. Moved the splitBucketID tests to the satellite/orders package.

Please describe the performance impact:
N/A

## Code Review Checklist (to be filled out by reviewer)
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: pkg/accounting/tally/tally.go [hunk 1]
```diff
@@ -142,9 +143,13 @@ func (t *Service) CalculateAtRestData(ctx context.Context) (latestTally time.Tim
 					bucketID := storj.JoinPaths(project, bucketName)
 
 					bucketTally := bucketTallies[bucketID]
+					projectID, err := uuid.Parse(project)
+					if err != nil {
+						return err
```
Comment (egonelbre) [inline pkg/accounting/tally/tally.go]: wrap error

File: pkg/accounting/tally/tally.go [hunk 2]
```diff
@@ -142,9 +143,13 @@ func (t *Service) CalculateAtRestData(ctx context.Context) (latestTally time.Tim
...
+					}
 					if bucketTally == nil {
 						bucketTally = &accounting.BucketTally{}
-						bucketTally.ProjectID = []byte(project)
+						bucketTally.ProjectID = projectID[:]
```
Comment (egonelbre) [inline pkg/accounting/tally/tally.go]: Why is ProjectID here not `uuid`?

File: satellite/orders/endpoint.go [hunk 3]
```diff
@@ -208,7 +209,11 @@ func (endpoint *Endpoint) Settlement(stream pb.Orders_SettlementServer) (err err
 		now := time.Now().UTC()
 		intervalStart := time.Date(now.Year(), now.Month(), now.Day(), now.Hour(), 0, 0, 0, now.Location())
 
-		if err := endpoint.DB.UpdateBucketBandwidthSettle(ctx, bucketID, orderLimit.Action, order.Amount, intervalStart); err != nil {
+		projectID, bucketName, err := SplitBucketID(bucketID)
```
Comment (egonelbre) [inline satellite/orders/endpoint.go]: UseSerialNumber should be returning project id and bucket name separately.

File: satellite/orders/orders_test.go [hunk 4]
```diff
@@ -174,3 +176,27 @@ func noLongTailRedundancy(planet *testplanet.Planet) uplink.RSConfig {
 	redundancy.SuccessThreshold = redundancy.MaxThreshold
 	return redundancy
 }
+
+func TestSpliteBucketID(t *testing.T) {
```
Comment (egonelbre) [inline satellite/orders/orders_test.go]: typo `TestSpliteBucketID`

File: satellite/orders/service.go [hunk 5]
```diff
@@ -204,7 +205,11 @@ func (service *Service) CreateGetOrderLimits(ctx context.Context, uplink *identi
 		return nil, Error.Wrap(err)
 	}
 
-	if err := service.updateBandwidth(ctx, bucketID, limits...); err != nil {
+	projectID, bucketName, err := SplitBucketID(bucketID)
```
Comment (egonelbre) [inline satellite/orders/service.go]: The bucket id shouldn't need to be split the argument should already be split.

File: satellite/orders/service.go [hunk 6]
```diff
@@ -664,27 +685,42 @@ func (service *Service) CreatePutRepairOrderLimits(ctx context.Context, repairer
...
 	intervalStart := time.Date(now.Year(), now.Month(), now.Day(), now.Hour(), 0, 0, 0, now.Location())
 
-	return service.orders.UpdateBucketBandwidthInline(ctx, bucketID, pb.PieceAction_PUT, amount, intervalStart)
+	return service.orders.UpdateBucketBandwidthInline(ctx, projectID, bucketName, pb.PieceAction_PUT, amount, intervalStart)
+}
+
+// SplitBucketID takes a bucketID, splits on /, and returns a projectID and bucketName
+func SplitBucketID(bucketID []byte) (projectID *uuid.UUID, bucketName []byte, err error) {
+	pathElements := bytes.Split(bucketID, []byte("/"))
+	bucketName = pathElements[1]
```
Comment (egonelbre) [inline satellite/orders/service.go]: This panics if there isn't enough elements.

File: pkg/accounting/tally/tally.go (hunk 2 above)
Comment (JessicaGreben) [reply pkg/accounting/tally/tally.go]: I attempted to change it to uuid here, but it became kind of messy so I decided to leave it at []byte

Comment (egonelbre) [reply pkg/accounting/tally/tally.go]: Add a note somewhere that this needs to be fixed.

File: satellite/orders/endpoint.go (hunk 3 above)
Comment (JessicaGreben) [reply satellite/orders/endpoint.go]: the serial number tables stores the bucketID so thats what it returns. I would have to split the bucketID

Comment (egonelbre) [reply satellite/orders/endpoint.go]: ah, right... I thought we already had split that as well

File: satellite/orders/service.go (hunk 5 above)
Comment (JessicaGreben) [reply satellite/orders/service.go]: bucketID gets passed in as an arg to this function

Comment (egonelbre) [reply satellite/orders/service.go]: Ah, meaning `CreateGetOrderLimits` should probably already get projectID and bucket name separately. But yeah, we can leave these for future.

**Pull request merged**